languages/csharp/exercises/concept/arrays/.docs/after.md

```diff
@@ -0,0 +1,104 @@
+Data structures that can hold zero or more elements are known as _collections_. An **array** is a collection that has a fixed size and whose elements must all be of the same type. Elements can be assigned to an array or retrieved from it using an index. C# arrays are zero-based, meaning that the first element's index is always zero:
+
+```csharp
```

Comment:
hmmm... just noticed that this is the `after` document, not the introduction one... 
why do we need re-introduce all concepts (including detailed  a syntax description) for an exercise the student just finished?

Reply:
It's also weird that this document contains more details than the `introduction` one...

Reply:
@maurelio1234 Well that is precisely what the `after.md` document should do. This comment to me seems to imply that you're not yet familiar with what the contents of the documents should be. Therefore, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

Reply:
Yes, I saw these documents before posting (not the video though) and a couple of other exercises... but it is still weird to me to give more details after then before.

I see the point of summarizing what was presented in the exercise (as it is said here: https://github.com/exercism/v3/blob/master/docs/concept-exercises.md#docsaftermd) but not of presenting the same thing again with more details... it seems too late for that... 

But again :) thanks for the pointers I will see the video.

Reply:
> but it is still weird to me to give more details after then before.

Well, we disagree :) The thing we want is for a student to learn a concept by themselves. This means providing them with the minimal amount of information necessary to solve the exercise. Once the student has demonstrated the mastery of the concepts by completing the exercise, the student is then ready to dig a bit deeper into the concept, for example by showing why things work a specific way or by linking to more in-depth tutorials.

Hopefully the video will help explain things.

Reply:
> Once the student has demonstrated the mastery of the concepts by completing the exercise, the student is then ready to dig a bit deeper into the concept, for example by showing why things work a specific way or by linking to more in-depth tutorials.

Oh yes, I see ;) It makes sense now ;)